core/src/main/scala/kafka/utils/Logging.scala

```diff
@@ -17,106 +17,96 @@
 
 package kafka.utils
 
-import org.apache.log4j.Logger
+import com.typesafe.scalalogging.{LazyLogging, Logger}
+import org.slf4j.{Marker, MarkerFactory}
 
-trait Logging {
-  val loggerName = this.getClass.getName
-  lazy val logger = Logger.getLogger(loggerName)
+import scala.util.{Failure, Success, Try}
 
-  protected var logIdent: String = null
+object Log4jControllerRegistration {
+  private val logger = Logger(this.getClass.getName)
 
-  // Force initialization to register Log4jControllerMBean
-  private val log4jController = Log4jController
+  val FatalMarker: Marker = MarkerFactory.getMarker("FATAL")
 
-  protected def msgWithLogIdent(msg: String) =
-    if (logIdent == null) msg else logIdent + msg
-
-  def trace(msg: => String): Unit = {
-    if (logger.isTraceEnabled())
-      logger.trace(msgWithLogIdent(msg))
-  }
-  def trace(e: => Throwable): Any = {
-    if (logger.isTraceEnabled())
-      logger.trace(logIdent,e)
+  val instance = Try {
+    val log4jController = Class.forName("kafka.utils.Log4jController").asInstanceOf[Class[Object]]
+    val instance = log4jController.newInstance()
+    CoreUtils.registerMBean(instance, "kafka:type=kafka.Log4jController")
+    instance
   }
-  def trace(msg: => String, e: => Throwable) = {
-    if (logger.isTraceEnabled())
-      logger.trace(msgWithLogIdent(msg),e)
-  }
-  def swallowTrace(action: => Unit) {
-    CoreUtils.swallow(logger.trace, action)
+
+  instance match {
+    case Success(_) => logger.info("Registered kafka:type=kafka.Log4jController MBean")
+    case Failure(_) => logger.info("Couldn't register kafka:type=kafka.Log4jController MBean")
   }
+}
 
-  def isDebugEnabled: Boolean = logger.isDebugEnabled
+trait Logging extends LazyLogging {
+  def loggerName: String = this.getClass.getName
```

Comment:
Hmm, how is this supposed to work? It doesn't seem like it's used anywhere? `LazyLogging` doesn't provide a way to override the logger name, it seems like one needs to override the `logger` itself:

```scala
trait LazyLogging {
  protected lazy val logger: Logger =
    Logger(LoggerFactory.getLogger(getClass.getName))
}
```

It doesn't look like `LazyLogging` buys us anything.

Reply:
It is used in `Cleaner`, `CleanerThread` and `LogCleaner`, however you're right that we shouldn't use this but create scala `Logger`s in these classes, and in this case `LazyLogging` won't buy us anything.

Reply:
Sorry, I wasn't clear. I meant that it's not used in `Logging` itself and hence the overrides would be ignored. I think we're on the same page.

Reply:
@viktorsomogyi looking at the code, what `Cleaner`, `CleanerThread` and `LogCleaner` are doing is a bit of an anti-pattern. So, let's drop the logger overrides.

Reply:
Nevermind my last comment. Unfortunately, we can't drop the overrides because the log4j config relies on that for the cleaner appender.